src/Symfony/Component/Validator/Constraints/Collection.php

```diff
@@ -87,4 +87,31 @@ protected function getCompositeOption()
     {
         return 'fields';
     }
+
+    private static function isFieldsOption($options): bool
+    {
+        if (!\is_array($options)) {
+            return false;
+        }
+
+        if ([] === $options) {
+            return true;
+        }
+
+        foreach ($options as $optionOrField) {
+            if ($optionOrField instanceof Constraint) {
+                return true;
+            }
+
+            if (!\is_array($optionOrField)) {
+                return false;
+            }
+
+            if ([] !== $optionOrField && !($optionOrField[0] ?? null) instanceof Constraint) {
```

Comment:
In theory this would still break constructions like these
```php
$c = new Collection(
    [
        'foo' => [  // implicit "new Required(...)"
            [
                new Type('string'),
            ],
        ],
    ],
);
$c = new Collection(
    [
        'foo' => [  // implicit "new Required(...)"
            'constraints' => [
                new Type('string'),
            ],
            'groups' => ['bar'],
        ],
    ],
);
```

Reply:
I am not sure I understand what you mean. Both examples where already invalid before, right?

Reply:
I meant that they originally worked before https://github.com/symfony/symfony/pull/53133

Reply:
Are you sure? How were both forms supposed to end up?

Reply:
Yes, both work on 6.4.0. That 0 index access seemed suspicous to me, so I tried to break it a bit.
(Note that I don't care about this working, but thought it worthwhile to point out)

These examples would turn into
```php
new Collection(
    fields: [
        'foo' => new Required([
            new Type('string'),
        ]),
    ],
);
// more explicit
new Collection(
    fields: [
        'foo' => new Required(
            options:  [
                'constraints' => [
                    new Type('string')
                ],
            ],
            groups: ['bar'],
        ),
    ],
);
```